### ISSUE TITLE
Android Vulkan: Project Diva Series, Little Blank Black Screen

### ISSUE THREAD

**Issue by TediExcaliber** [hrydgard/ppsspp#10107]
Now I using v1.4.2-739-g3dd08b84f and when playing PD Series with vulkan some part of screen is going black... Yeah I know, it just small but sometime disturbing 😁 i dont know direcly what build version is starts this bug, but this bug is coming since I using builld 728

This bug happen in PD Extend and 2nd and maybe PD1 too... In backend Vulkan, i just try this on my Phone, cuz i dont have PC and I have try to changing some setting but not work.

Here the screenshot

http://imgur.com/0V7KKGo > OpenGL
http://imgur.com/N1ZscMb > Vulkan

**Comment by kurayami6** [comment]
I can confirm this also happens in Windows. Except I only get it on the loading screens before and after songs.

**Comment by TediExcaliber** [comment]
Update: This bug also happen in some of my games

Like Dragonball Z Budokai AR, DW Strikeforce, and PoP Revelations

**Comment by hrydgard** [comment]
@TediExcaliber What phone are you on?

**Comment by TediExcaliber** [comment]
Iam using Xiaomi Redmi 4X with Android N

Pada tanggal 14 Nov 2017 5.01 PM, "Henrik Rydgård" <notifications@github.com>
menulis:

**Comment by hrydgard** [comment]
Can you go back a few builds and try to figure out when this started to happen?

**Comment by TediExcaliber** [comment]
Sorry sir, i cant do that, i have limit internet data, cuz, all of place in
my village dont have wi fi 🙏

Pada tanggal 14 Nov 2017 5.14 PM, "Henrik Rydgård" <notifications@github.com>
menulis:

**Comment by hrydgard** [comment]
Oh, okay. I tend to take fast internet for granted, sorry :)

**Comment by unknownbrackets** [comment]
Could you try exporting a GE debugger dump on PC - or ask someone to help?  That's assuming it still happens in the latest git build.

To do this, open the game and select Debug -> GE debugger..., then when it's displaying the scene, press Record in the top right.  After a second or so, it'll finish and save a trace of the drawing activity.

After that, check the memstick/PSP/SYSTEM/DUMP folder and it'll have created a file named something like "ULES12345_0000.ppdmp".  You can zip that and then drag and drop it into a reply here.

-[Unknown]

**Comment by kurayami6** [comment]
Is this what you need? Hope I did it correctly.

[ULJM05681_0002.zip](https://github.com/hrydgard/ppsspp/files/1547352/ULJM05681_0002.zip)

**Comment by unknownbrackets** [comment]
If I disable clipping (which clamps depth), it leaves a solid black line underneath.

 * It starts out clearing depth to 0, which actually clears to 0.015625 (1/64.)
 * Next, it draws some UI across the entire bottom (including the black box area.)  The depth value is, after transform, 65527.445313, which is rounded to 65527.453125.
 * When it goes to draw the full screen image on top, that depth value is constant at the bottom 6 pixels, and the rest has 0.015625 - no other depth values.
 * The full screen image is drawn with src >= dst depth testing, and depth clamping enabled.
 * The incoming source depth value is 65527.445313 for all vertices (it uses two full triangles - not a strip or fan.)
 * The resulting depth values range from:
   * Top rightish area: 65527.437500
   * Bottom leftish area: 65527.453125 (equal to the value actually there)
   * Middle between those triangles: 65527.421875 (less than the value actually there)

Depth clamping effects this, because it slightly alters the rounding.  Since the bottom left is the correct value, the equality test passes.  But the other corners aren't so lucky.

No pixel rounding is being performed.  All three depth hacks (GPU_SCALE_DEPTH_FROM_24BIT_TO_16BIT, GPU_ROUND_FRAGMENT_DEPTH_TO_16BIT, and GPU_ROUND_DEPTH_TO_16BIT) do fix it.

In OpenGL (non-accurate depth), depth is initially cleared to 0 (so it doesn't have the slight rounding issue), and the first draw sets depth to 65527.445313, which is the same value it results in.

So basically, the verts are being rounded differently since they don't target a representable depth value directly, in the accurate depth path (the one that doesn't incorrectly clip depth outside the viewport, and does clip it based on the depth range registers.)

-[Unknown]

**Comment by leopard20** [comment]
@hrydgard 
I think the issue is already fixed. I have completed the game and haven't encountered this issue.
Or maybe it's a driver bug.

**Comment by kurayami6** [comment]
It's absolutely not fixed. I'm using PPSSPP version 1.6.3-70-ga08fb8bd2 with Windows 7 and it's still happening in Diva 2nd.

**Comment by leopard20** [comment]
Well it is fixed for me (wasn't even there to begin with).

**Comment by leopard20** [comment]
Exactly the same scene as above, Vulkan backend, Android 7.0, Mi 5s Plus:

![screenshot_2018-06-07-04-27-15-276_org ppsspp ppsspp](https://user-images.githubusercontent.com/16115820/41071038-3a4cb780-6a0b-11e8-83c2-4fd6df3a4219.png)

**Comment by unknownbrackets** [comment]
It may be card/driver specific, since it's a depth rounding issue.

-[Unknown]

**Comment by leopard20** [comment]
I'm pretty sure I didn't see it on my Windows PC either (Nvidia GTX 760m, driver version ~380-ish I think). But I'll check later to make sure.

**Comment by leopard20** [comment]
@TediExcaliber 
IIRC, the Snapdragon 435 in Redmi 4x doesn't support Vulkan officially. Have you flashed the Vulkan injector on XDA? It may be related to that.

**Comment by unknownbrackets** [comment]
It's possible the depth clamp behavior described in #12058 will fix this (at least it appears to in the software renderer.)

-[Unknown]

**Comment by Gamemulatorer** [comment]
How about this now after https://github.com/hrydgard/ppsspp/commit/3ff400e40e122dcba771d32b8a064a3b42e7e371

**Comment by unknownbrackets** [comment]
As with the other issues, clipping against negative Z helped this but only works on a subset of devices right now.  It's still broken on Mali devices.

-[Unknown]

**Comment by Crisanto47** [comment]
An update ge dump please I will test in my redmi 4x phone

**Comment by Crisanto47** [comment]
> As with the other issues, clipping against negative Z helped this but only works on a subset of devices right now. It's still broken on Mali devices.
> 
> -[Unknown]

This is similar to this probably https://github.com/hrydgard/ppsspp/issues/12295
